q2_longitudinal/_utilities.py

```diff
@@ -400,40 +391,84 @@ def _regplot_subplots_from_dataframe(state_column, metric, metadata,
 
 
 def _control_chart_subplots(state_column, metric, metadata, group_column,
-                            ci=95, palette='Set1', plot_control_limits=True,
-                            xtick_interval=None):
+                            individual_id_column, ci=95, palette='Set1',
+                            plot_control_limits=True, xtick_interval=None,
+                            yscale='linear', spaghetti='no'):
 
     groups = metadata[group_column].unique()
-    states = metadata[state_column].unique()
-    chart, axes = plt.subplots(len(groups) + 1, figsize=(6, 18))
+    states = sorted(metadata[state_column].unique())
+    fig_count = len(groups) + 1
+    fig_height = fig_count * 6
+    chart, axes = plt.subplots(fig_count, figsize=(6, fig_height))
 
     # determine x tick interval: autoscale so that ≤ 30 labels appear
     xtick_interval = _set_xtick_interval(xtick_interval, states)
 
-    # plot all groups together, compare variances
-    c, global_mean, global_std = _control_chart(
-        state_column, metric, metadata, group_column, ci=ci, palette=palette,
-        plot_control_limits=plot_control_limits, ax=axes[0],
-        xtick_interval=xtick_interval)
-    c.set_title('Group volatility comparison plot')
-    c = _set_xticks(c, metadata, state_column, states, xtick_interval)
-
     # plot individual groups' control charts
     colors = cycle(sns.color_palette(palette, n_colors=len(groups)))
-    for num in range(len(groups)):
-        group = groups[num]
-        group_md = metadata[metadata[group_column] == group]
+    cmap = {}
```

Comment:
try:
```python
cmap = dict(zip(groups, sns.color_palette(palette, n_colors=len(groups))))
```

Reply:
👍

Reply:
actually... this appears to not work (the colors are not ordered correctly). I already cycle through the colors in the following `for` loop, so I think it makes sense to just assign the colors in there (making sure the same color order is used) rather than dict comprehension.

Reply:
oh sorry, that should be a `collections.OrderedDict` to make the order deterministic. And then ideally you would index that via `group` wherever you previously used `color`. I was hoping we could reduce the number of active "iterables" in the loop.

Reply:
got it, thanks for pointing out the active iterables issue  — `dict` is fine, I just need to call `cmap` from within the loop instead of cycling the colors! :man_facepalming: